packages/workflow/src/Extensions/DateExtensions.ts

```diff
@@ -154,54 +150,51 @@ function isBetween(
 	return secondDate > date && date > firstDate;
 }
 
-function isDst(date: Date | DateTime): boolean {
-	if (isDateTime(date)) {
-		return date.isInDST;
-	}
-	return DateTime.fromJSDate(date).isInDST;
+function isDst(date: Date | LuxonDateTime): boolean {
+	return toLuxonDateTime(date).isInDST;
 }
 
-function isInLast(date: Date | DateTime, extraArgs: unknown[]): boolean {
+function isInLast(date: Date | LuxonDateTime, extraArgs: unknown[]): boolean {
 	const [durationValue = 0, unit = 'minutes'] = extraArgs as [number, DurationUnit];
 
-	const dateInThePast = DateTime.now().minus(generateDurationObject(durationValue, unit));
+	const dateInThePast = LuxonDateTime.now().minus(generateDurationObject(durationValue, unit));
 	let thisDate = date;
-	if (!isDateTime(thisDate)) {
-		thisDate = DateTime.fromJSDate(thisDate);
+	if (!isLuxonDateTime(thisDate)) {
+		thisDate = LuxonDateTime.fromJSDate(thisDate);
 	}
-	return dateInThePast <= thisDate && thisDate <= DateTime.now();
+	return dateInThePast <= thisDate && thisDate <= LuxonDateTime.now();
 }
 
 const WEEKEND_DAYS: WeekdayNumbers[] = [6, 7];
-function isWeekend(date: Date | DateTime): boolean {
-	const { weekday } = isDateTime(date) ? date : DateTime.fromJSDate(date);
+function isWeekend(date: Date | LuxonDateTime): boolean {
+	const { weekday } = toLuxonDateTime(date);
 	return WEEKEND_DAYS.includes(weekday);
 }
 
-function minus(date: Date | DateTime, extraArgs: unknown[]): Date | DateTime {
-	if (isDateTime(date) && extraArgs.length === 1) {
-		return date.minus(extraArgs[0] as DurationLike);
-	}
+function minus(date: Date | LuxonDateTime, extraArgs: unknown[]): string {
+	const [arg] = extraArgs as [DurationLike];
 
-	const [durationValue = 0, unit = 'minutes'] = extraArgs as [number, DurationUnit];
+	if (isLuxonDateTime(date) && arg) return date.minus(arg).toISO();
 
-	if (isDateTime(date)) {
-		return date.minus(generateDurationObject(durationValue, unit)).toJSDate();
-	}
-	return DateTime.fromJSDate(date).minus(generateDurationObject(durationValue, unit)).toJSDate();
+	const duration = run(() => {
+		const [durationValue = 0, unit = 'minutes'] = extraArgs as [number, DurationUnit];
+		return generateDurationObject(durationValue, unit);
+	});
```

Comment:
Does this need to be wrapped in a function?

Reply:
I was expecting this :)

So I'm growing fond of Rust's [block expressions](https://doc.rust-lang.org/reference/expressions/block-expr.html) and I really like how they help with structuring code and readability - making it clear that a number of lines are only there to generate a `duration`. But I also realize it's not traditional JS and the extra function does incur a small cost. Let me know if you'd prefer otherwise and I'll revert!

Reply:
I'm also very fond block expressions. I'm personally fine with even if there's a very small runtime cost but other might not be. I think either remove it or move the run function itself somewhere else since it feels a bit weird for it only be in that one file.

Reply:
Agree, removed! The readability gain is pretty minor in this case.